prow/github/client.go

```diff
@@ -1312,14 +1316,55 @@ func (c *Client) CreateReview(org, repo string, number int, r DraftReview) error
 	return err
 }
 
+// prepareReviewersBody separates reviewers from team_reviewers and prepares a map
+// {
+//   "reviewers": [
+//     "octocat",
+//     "hubot",
+//     "other_user"
+//   ],
+//   "team_reviewers": [
+//     "justice-league"
+//   ]
+// }
+//
+// https://developer.github.com/v3/pulls/review_requests/#create-a-review-request
+func prepareReviewersBody(logins []string, org string) (map[string][]string, error) {
+	body := map[string][]string{}
+	var errors []error
+	for _, login := range logins {
+		mat := teamRe.FindStringSubmatch(login)
+		if mat == nil {
+			if _, exists := body["reviewers"]; !exists {
+				body["reviewers"] = []string{}
+			}
+			body["reviewers"] = append(body["reviewers"], login)
+		} else if mat[1] == org {
+			if _, exists := body["team_reviewers"]; !exists {
+				body["team_reviewers"] = []string{}
+			}
+			body["team_reviewers"] = append(body["team_reviewers"], mat[2])
+		} else {
+			errors = append(errors, fmt.Errorf("team %s is not part of %s org", login, org))
+		}
+	}
+	return body, errorutil.NewAggregate(errors...)
+}
+
 func (c *Client) tryRequestReview(org, repo string, number int, logins []string) (int, error) {
 	c.log("RequestReview", org, repo, number, logins)
 	var pr PullRequest
+	body, err := prepareReviewersBody(logins, org)
+	if len(body) == 0 {
+		return 400, err
+	} else if err != nil {
+		c.logger.WithError(err).Warningf("Could not process all logins in %s", logins)
```

Comment:
Logging the error isn't sufficient here because the plugin needs to be able to respond to this case with a comment that lists the unrecognized users and teams.  We ultimately need to return a unique error type that the plugin can catch to distinguish the invalid input from a client error.

However you want to achieve this is fine, the important thing is that we are able to respond to invalid usage of the command with a comment. This could get overly complicated if there are errors preparing the body and also errors making the request (especially because of the weird individual retry logic we have to do) so if you want to switch this back to not being best effort and just returning an error immediately upon invalid input without making a request that is fine by me. 
Sorry for flip flopping on this. I forgot about the individual retry logic and the `MissingUsers` and `ExtraUsers` errors.

Reply:
You know, actually I think the retry loop in RequestReview will perfectly handle comments if I just adapt the answer of tryRequestReview...